ipmininet/link.py

```diff
@@ -150,7 +154,7 @@ def _set_ip(self, ip, prefixLen=None):
                     addr = ip_interface(u'%s/%s' % (addr, prefixLen))
                 else:
                     # no prefixLen defaults to full /128 or /32
-                    addr = ip_interface(unicode(addr))
+                    addr = ip_interface(str(addr))
```

Comment:
Did you test this in py2?

IIRC the [python2 ipaddress module backport](https://pypi.org/project/ipaddress/) will straight out [refuse non-unicode input strings](https://github.com/phihag/ipaddress/blob/master/README.md).

Reply:
All the code was tested on Python 2.7, 3.5, 3.6 and 3.7.
This line is not covered by tests, but you have a similar modification in ipmininet/utils.py, line 77 which is covered by the tests.

The "str" is not the python2 "str". It is imported from "builtins" (which is installed with the future package) at the top of the file.
This function consistently returns a unicode string no matter the python version.

If you still need to be conviced, this code works in python2.7 and the latest version of ipaddress.
```python
import ipaddress
from builtins import str
ipaddress.ip_address(str(b"1.1.1.1"))
```
while the following doesn't
```
import ipaddress
ipaddress.ip_address(str(b"1.1.1.1"))
˘˘˘

Reply:
I did not notice the str override through import.
What a nasty/clever monkey patch.